idea/idea-frontend-api/src/org/jetbrains/kotlin/idea/frontend/api/calls/KtCall.kt

```diff
@@ -53,12 +55,48 @@ public class KtVariableWithInvokeFunctionCall(
     override val targetFunction: KtCallTarget
 ) : KtDeclaredFunctionCall()
 
+/**
+ * Represents a direct function call with arguments
+ */
+public sealed class KtCallWithArguments : KtDeclaredFunctionCall() {
+    public abstract val argumentMapping: LinkedHashMap<KtValueArgument, KtValueParameterSymbol>
+}
+
 /**
  * Simple function call, e.g.,
  *
  * x.toString() // function call
  */
-public data class KtFunctionCall(override val targetFunction: KtCallTarget) : KtDeclaredFunctionCall()
+public class KtFunctionCall(
+    override val argumentMapping: LinkedHashMap<KtValueArgument, KtValueParameterSymbol>,
+    override val targetFunction: KtCallTarget
+) : KtCallWithArguments()
+
+/**
+ * Annotation call, e.g.,
+ *
+ * @Retention(AnnotationRetention.SOURCE) // annotation call
+ * annotation class Ann
+ */
+public class KtAnnotationCall(
+    override val argumentMapping: LinkedHashMap<KtValueArgument, KtValueParameterSymbol>,
+    override val targetFunction: KtCallTarget
+) : KtCallWithArguments()
+// TODO: Add other properties, e.g., useSiteTarget
+
+/**
+ * Delegated constructor call, e.g.,
+ *
+ * open class A(a: Int)
+ * class B(b: Int) : A(b) { // delegated constructor call (isThis = false)
+ *   constructor() : this(1) // delegated constructor call (isThis = true)
+ * }
+ */
+public class KtDelegatedConstructorCall(
+    override val argumentMapping: LinkedHashMap<KtValueArgument, KtValueParameterSymbol>,
+    override val targetFunction: KtCallTarget,
+    public val isThis: Boolean
```

Comment:
What about `isSuper`?

Reply:
That should just be the opposite of `isThis`; they're mutually exclusive.

Reply:
Oh.... so the question should go to the tree design, like, why do we have this one: https://github.com/JetBrains/kotlin/blob/master/compiler/fir/tree/gen/org/jetbrains/kotlin/fir/expressions/FirDelegatedConstructorCall.kt#L28

Reply:
See impl: https://github.com/JetBrains/kotlin/blob/922aad686552b416539b1d36f17815c7fe92d7b5/compiler/fir/tree/gen/org/jetbrains/kotlin/fir/expressions/impl/FirDelegatedConstructorCallImpl.kt#L31-L33 😄 

Ok I'll add the same here.

Reply:
Duh 🤦

Reply:
Maybe it is better introduce an enum here? As isThis/isSuper always have opposite values. Smth like `KtDelegatedConstcutorKind { SUPER_CALL, THIS_CALL }`

Reply:
Good suggestion, I'm changing to an enum.

Reply:
Maybe out of scope of this PR, but I wonder if it's good to change FIR tree as well to have such enum?

Or, we can introduce similar properties here, like:
```kt
public class KtDelegatedConstructorCall(
    ...
    private val kind: KtDelegatedConstructorCallKind
) : KtCallWithArguments() {
  public val isThis: Boolean
      get() = kind == THIS_CALL
  public val isSuper: Boolean
      get() = kind == SUPER_CALL
}
```
:)